sequencers/modeshape-sequencer-epub/src/main/java/org/modeshape/sequencer/epub/EpubMetadata.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * ModeShape (http://www.modeshape.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modeshape.sequencer.epub;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import javax.xml.namespace.NamespaceContext;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * Utility for extracting Metadata from EPUB format.
+ */
+public class EpubMetadata {
+
+    static final String[] MIME_TYPE_STRINGS = { "application/epub+zip" };
+
+    // The XML namespace for the Dublin Core schema.
+    static final String DUBLIN_CORE_PREFIX = "dc";
+    static final String DUBLIN_CORE_URI = "http://purl.org/dc/elements/1.1/";
+
+    private String title;
+    private String creator;
+    private String contributor;
+    private String language;
+    private String identifier;
+    private String subject;
+    private String description;
+    private String publisher;
+    private String date;
+
+    private InputStream in;
+    private DataInput din;
+
+    /*
+     * Check that given file is supported by this sequencer.
+     */
+    public boolean check() {
+        try {
+            ZipInputStream zip = new ZipInputStream(in);
+            ZipEntry entry = null;
+            while ((entry = zip.getNextEntry()) != null) {
+                if (entry.getName().endsWith("/content.opf")) {
```

Comment:
I don't believe the EPUB spec requires the rootfile to be named "content.opf"

The name of the rootfile (or rootfiles) is definied in META-INF/container.xml  (see http://www.idpf.org/epub/301/spec/epub-ocf.html#sec-container-metainf-container.xml )